src/cmlibs/utils/zinc/group.py

```diff
@@ -388,3 +388,52 @@ def group_evaluate_representative_point(group, coordinate_field,
     elif nodeset_group:
         return evaluate_field_nodeset_mean(coordinate_field, nodeset_group)
     return None
+
+
+def groups_have_same_local_contents(group1, group2):
+    """
+    Determine if two groups have the same contents in the local/owning region only.
+    Empty and non-existent mesh/nodeset groups are considered to be the same.
+    :param group1: Zinc group.
+    :param group2: Zinc group from same region as group1.
+    :return: True if same contents, otherwise False.
+    """
+    fieldmodule = group1.getFieldmodule()
+    assert fieldmodule.getRegion() == group2.getFieldmodule().getRegion()
```

Comment:
I don't think this should assert, just return False.

Reply:
Fixed.